src/corehost/cli/apphost/bundle/bundle_runner.h

```diff
@@ -50,10 +52,9 @@ namespace bundle
         void extract_file(file_entry_t* entry);
 
         FILE* m_bundle_stream;
+        header_t* m_header;
```

Comment:
What is destroying `m_header`?  The memory leak is still here.  A few options here, in order from the most preferred to the least preferred:

- Declare this inline as `header_t m_header`. No need to allocate a new struct from the heap. Goes away as soon as `bundle_runner_t` goes away.
- Make this a `std::unique_ptr<header_t>` instead.
- Have a `delete m_header` in `bundle_runner_t::~bundle_runner_t()`.

Reply:
I added a destructor to the `bundle_runner_t`.
I did not inline the manifest, because when files are directly processed from the bundle, the manifest will survive beyond the `bundle_runner`  object.
I also wanted to keep the interface of `header_t` similar to `manifest_t` and the managed code in `HostModel`. Thanks.